tests/projects/test_projects_cli.py

```diff
@@ -29,3 +29,36 @@ def test_run_git():
             res = invoke_cli_runner(cli.run, [GIT_PROJECT_URI, "--no-conda", "-P", "alpha=0.5"])
             assert "python train.py 0.5 0.1" in res.output
             _assert_succeeded(res.output)
+
+GIT_SUBDIR_URI = "https://github.com/juntai-zheng/mlflow-git-features.git"
+
+
+def test_git_subdirectories():
```

Comment:
Thanks for the tests, ideally these would ideally be unit tests in `test_projects.py` that call the project-fetching helper function - while integration tests are helpful in simulating the user workflow, in general it's preferable to have fewer integration tests & more unit tests (goes back to the testing pyramid idea) since the latter are (a) quicker to run and (b) easier to debug, since we know exactly what logic failed